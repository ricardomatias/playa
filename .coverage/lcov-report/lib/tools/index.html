<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for lib/tools</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> lib/tools
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">91.08% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>296/325</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">85% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>102/120</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">84.75% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>50/59</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">90.76% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>275/303</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line high'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="choose.js"><a href="choose.js.html">choose.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	</tr>

<tr>
	<td class="file high" data-value="distance.js"><a href="distance.js.html">distance.js</a></td>
	<td data-value="97.37" class="pic high"><div class="chart"><div class="cover-fill" style="width: 97%;"></div><div class="cover-empty" style="width:3%;"></div></div></td>
	<td data-value="97.37" class="pct high">97.37%</td>
	<td data-value="38" class="abs high">37/38</td>
	<td data-value="94.44" class="pct high">94.44%</td>
	<td data-value="18" class="abs high">17/18</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	<td data-value="97.14" class="pct high">97.14%</td>
	<td data-value="35" class="abs high">34/35</td>
	</tr>

<tr>
	<td class="file high" data-value="distribute.js"><a href="distribute.js.html">distribute.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="24" class="abs high">24/24</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="20" class="abs high">20/20</td>
	</tr>

<tr>
	<td class="file high" data-value="euclidean.js"><a href="euclidean.js.html">euclidean.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="20" class="abs high">20/20</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="18" class="abs high">18/18</td>
	</tr>

<tr>
	<td class="file high" data-value="friendly.js"><a href="friendly.js.html">friendly.js</a></td>
	<td data-value="97.56" class="pic high"><div class="chart"><div class="cover-fill" style="width: 97%;"></div><div class="cover-empty" style="width:3%;"></div></div></td>
	<td data-value="97.56" class="pct high">97.56%</td>
	<td data-value="82" class="abs high">80/82</td>
	<td data-value="92.59" class="pct high">92.59%</td>
	<td data-value="27" class="abs high">25/27</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="13" class="abs high">13/13</td>
	<td data-value="97.4" class="pct high">97.4%</td>
	<td data-value="77" class="abs high">75/77</td>
	</tr>

<tr>
	<td class="file empty" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="0" class="pic empty"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	</tr>

<tr>
	<td class="file high" data-value="interval.js"><a href="interval.js.html">interval.js</a></td>
	<td data-value="88.89" class="pic high"><div class="chart"><div class="cover-fill" style="width: 88%;"></div><div class="cover-empty" style="width:12%;"></div></div></td>
	<td data-value="88.89" class="pct high">88.89%</td>
	<td data-value="36" class="abs high">32/36</td>
	<td data-value="85" class="pct high">85%</td>
	<td data-value="20" class="abs high">17/20</td>
	<td data-value="87.5" class="pct high">87.5%</td>
	<td data-value="8" class="abs high">7/8</td>
	<td data-value="91.43" class="pct high">91.43%</td>
	<td data-value="35" class="abs high">32/35</td>
	</tr>

<tr>
	<td class="file high" data-value="random.js"><a href="random.js.html">random.js</a></td>
	<td data-value="93.33" class="pic high"><div class="chart"><div class="cover-fill" style="width: 93%;"></div><div class="cover-empty" style="width:7%;"></div></div></td>
	<td data-value="93.33" class="pct high">93.33%</td>
	<td data-value="15" class="abs high">14/15</td>
	<td data-value="80" class="pct high">80%</td>
	<td data-value="5" class="abs high">4/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="92.86" class="pct high">92.86%</td>
	<td data-value="14" class="abs high">13/14</td>
	</tr>

<tr>
	<td class="file medium" data-value="ring.js"><a href="ring.js.html">ring.js</a></td>
	<td data-value="77.27" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 77%;"></div><div class="cover-empty" style="width:23%;"></div></div></td>
	<td data-value="77.27" class="pct medium">77.27%</td>
	<td data-value="22" class="abs medium">17/22</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="12" class="abs high">10/12</td>
	<td data-value="40" class="pct low">40%</td>
	<td data-value="5" class="abs low">2/5</td>
	<td data-value="77.27" class="pct medium">77.27%</td>
	<td data-value="22" class="abs medium">17/22</td>
	</tr>

<tr>
	<td class="file high" data-value="roll.js"><a href="roll.js.html">roll.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="9" class="abs high">9/9</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	</tr>

<tr>
	<td class="file high" data-value="rotate.js"><a href="rotate.js.html">rotate.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	</tr>

<tr>
	<td class="file high" data-value="shuffle.js"><a href="shuffle.js.html">shuffle.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="12" class="abs high">12/12</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="9" class="abs high">9/9</td>
	</tr>

<tr>
	<td class="file medium" data-value="time.js"><a href="time.js.html">time.js</a></td>
	<td data-value="71.93" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 71%;"></div><div class="cover-empty" style="width:29%;"></div></div></td>
	<td data-value="71.93" class="pct medium">71.93%</td>
	<td data-value="57" class="abs medium">41/57</td>
	<td data-value="67.86" class="pct medium">67.86%</td>
	<td data-value="28" class="abs medium">19/28</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="10" class="abs medium">5/10</td>
	<td data-value="71.43" class="pct medium">71.43%</td>
	<td data-value="56" class="abs medium">40/56</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Feb 20 2019 15:09:45 GMT+0100 (Central European Standard Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
<script src="../../block-navigation.js"></script>
</body>
</html>
