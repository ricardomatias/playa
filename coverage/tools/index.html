
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for tools</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../index.html">All files</a> tools</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">94.43% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>458/485</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">93.25% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>152/163</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">96% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>48/50</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">94.43% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>458/485</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line high'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="choose.ts"><a href="choose.ts.html">choose.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="29" class="abs high">29/29</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="12" class="abs high">12/12</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="29" class="abs high">29/29</td>
	</tr>

<tr>
	<td class="file high" data-value="distance.ts"><a href="distance.ts.html">distance.ts</a></td>
	<td data-value="93.54" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 93%"></div><div class="cover-empty" style="width: 7%"></div></div>
	</td>
	<td data-value="93.54" class="pct high">93.54%</td>
	<td data-value="31" class="abs high">29/31</td>
	<td data-value="81.81" class="pct high">81.81%</td>
	<td data-value="11" class="abs high">9/11</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="93.54" class="pct high">93.54%</td>
	<td data-value="31" class="abs high">29/31</td>
	</tr>

<tr>
	<td class="file high" data-value="euclidean.ts"><a href="euclidean.ts.html">euclidean.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="22" class="abs high">22/22</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="22" class="abs high">22/22</td>
	</tr>

<tr>
	<td class="file high" data-value="event.ts"><a href="event.ts.html">event.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="66" class="abs high">66/66</td>
	<td data-value="92" class="pct high">92%</td>
	<td data-value="25" class="abs high">23/25</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="66" class="abs high">66/66</td>
	</tr>

<tr>
	<td class="file high" data-value="index.ts"><a href="index.ts.html">index.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	</tr>

<tr>
	<td class="file high" data-value="interval.ts"><a href="interval.ts.html">interval.ts</a></td>
	<td data-value="92.45" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 92%"></div><div class="cover-empty" style="width: 8%"></div></div>
	</td>
	<td data-value="92.45" class="pct high">92.45%</td>
	<td data-value="53" class="abs high">49/53</td>
	<td data-value="89.47" class="pct high">89.47%</td>
	<td data-value="19" class="abs high">17/19</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	<td data-value="92.45" class="pct high">92.45%</td>
	<td data-value="53" class="abs high">49/53</td>
	</tr>

<tr>
	<td class="file high" data-value="map-repeat.ts"><a href="map-repeat.ts.html">map-repeat.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	</tr>

<tr>
	<td class="file high" data-value="midi.ts"><a href="midi.ts.html">midi.ts</a></td>
	<td data-value="96.72" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 96%"></div><div class="cover-empty" style="width: 4%"></div></div>
	</td>
	<td data-value="96.72" class="pct high">96.72%</td>
	<td data-value="122" class="abs high">118/122</td>
	<td data-value="91.3" class="pct high">91.3%</td>
	<td data-value="46" class="abs high">42/46</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	<td data-value="96.72" class="pct high">96.72%</td>
	<td data-value="122" class="abs high">118/122</td>
	</tr>

<tr>
	<td class="file high" data-value="next.ts"><a href="next.ts.html">next.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="17" class="abs high">17/17</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="17" class="abs high">17/17</td>
	</tr>

<tr>
	<td class="file low" data-value="ping.ts"><a href="ping.ts.html">ping.ts</a></td>
	<td data-value="13.33" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 13%"></div><div class="cover-empty" style="width: 87%"></div></div>
	</td>
	<td data-value="13.33" class="pct low">13.33%</td>
	<td data-value="15" class="abs low">2/15</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="13.33" class="pct low">13.33%</td>
	<td data-value="15" class="abs low">2/15</td>
	</tr>

<tr>
	<td class="file high" data-value="random.ts"><a href="random.ts.html">random.ts</a></td>
	<td data-value="94.93" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 94%"></div><div class="cover-empty" style="width: 6%"></div></div>
	</td>
	<td data-value="94.93" class="pct high">94.93%</td>
	<td data-value="79" class="abs high">75/79</td>
	<td data-value="94.44" class="pct high">94.44%</td>
	<td data-value="18" class="abs high">17/18</td>
	<td data-value="93.75" class="pct high">93.75%</td>
	<td data-value="16" class="abs high">15/16</td>
	<td data-value="94.93" class="pct high">94.93%</td>
	<td data-value="79" class="abs high">75/79</td>
	</tr>

<tr>
	<td class="file high" data-value="rotate.ts"><a href="rotate.ts.html">rotate.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="12" class="abs high">12/12</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="12" class="abs high">12/12</td>
	</tr>

<tr>
	<td class="file high" data-value="shuffle.ts"><a href="shuffle.ts.html">shuffle.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="13" class="abs high">13/13</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="13" class="abs high">13/13</td>
	</tr>

<tr>
	<td class="file high" data-value="zigzag.ts"><a href="zigzag.ts.html">zigzag.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="22" class="abs high">22/22</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="22" class="abs high">22/22</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at 2024-10-16T10:39:20.013Z
            </div>
        <script src="../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../sorter.js"></script>
        <script src="../block-navigation.js"></script>
    </body>
</html>
    