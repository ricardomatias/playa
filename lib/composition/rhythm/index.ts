export { createFreeRhythm as free } from './free';
export { createTurnRhythm as turn } from './turn';
export { createGrid } from './grid';
export { Presets } from './presets';
