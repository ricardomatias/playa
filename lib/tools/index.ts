export { choose, chooseMany } from './choose';
export { default as shuffle } from './shuffle';
export { default as rotate } from './rotate';
export { default as distance } from './distance';
export * from './interval';
export { friendly } from './friendly';
export * as Euclidean from './euclidean';
export { default as Sequence } from './sequence';
export { default as Random } from './random';

export * as Rhythm from './rhythm';
export * as Midi from './midi';

export * from './ping';
export { distribute, roll } from '@ricardomatias/roll';

