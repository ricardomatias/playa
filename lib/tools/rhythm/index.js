export { default as generateFreeRhythm } from './free';
export { default as generateTurnRhythm } from './turn';
export { default as createGrid } from './grid';
