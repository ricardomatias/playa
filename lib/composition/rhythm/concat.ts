import * as R from 'ramda';
import { Event } from '../../core';
import { mapStartToEvent } from '../../tools';
import { isUndefined } from '../../utils/types-guards';

/**
* Concats multiple rhythms ({@link Event}[]) together with the correct *time* and *next*
* @function concat
* @memberof Tools.Rhythm
*
* @param {Array<Array<Event>>} events
* @return {Array<Event>}
*/
export function concat(...events: Event[][]): Event[] {
	const start = events[0];

	if (isUndefined(start)) return [];

	let pattern = start;

	for (let index = 1; index < events.length; index++) {
		const curr = events[index];
		const start = R.last(pattern)?.next || 0;

		pattern = pattern.concat(R.map(mapStartToEvent(start), curr));
	}

	return pattern;
}
