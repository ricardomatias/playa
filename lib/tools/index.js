/**
 * The Toolshed
 * @namespace {module} Tools
 */
export { choose, chooseMany } from './choose';
export { default as shuffle } from './shuffle';
export { default as rotate } from './rotate';
export { default as distance } from './distance';
export { default as interval } from './interval';
export { default as friendly } from './friendly';
export { default as euclidean } from './euclidean';
export { default as Sequence } from './sequence';
export { default as Random } from './random';

import * as Time from './time';
import * as Rhythm from './rhythm';
import * as Midi from './midi';

export * from './ping';
export * from '@ricardomatias/roll';

export {
	Time,
	Rhythm,
	Midi,
};
