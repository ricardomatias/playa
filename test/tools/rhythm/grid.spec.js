import { createGrid } from '../../../lib/tools/rhythm';

describe('#createGrid', () => {
	it('should create a 8n grid', () => {
		const grid = createGrid(1920, 8);

		expect(grid).toHaveLength(16);
	});

	it('should create a 4n grid', () => {
		const grid = createGrid(3840, 4);

		expect(grid).toHaveLength(16);
	});

	it('should create a 4n grid - snapshot', () => {
		const grid = createGrid(960, 4);

		expect(grid).toHaveLength(4);
		expect(grid).toMatchSnapshot();
	});
});
