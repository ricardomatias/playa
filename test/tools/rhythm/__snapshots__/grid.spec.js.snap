// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#createGrid should create a 4n grid - snapshot 1`] = `
 [
  Object {
    "res": Array [
      "4nt",
      "4n",
    ],
    "time": 0,
  },
  Object {
    "res": Array [
      "4nt",
    ],
    "time": 320,
  },
  Object {
    "res": Array [
      "4n",
    ],
    "time": 480,
  },
  Object {
    "res": Array [
      "4nt",
    ],
    "time": 640,
  },
]
`;
